cli/deployment/config.go

```diff
@@ -0,0 +1,450 @@
+package deployment
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/coreos/go-oidc/v3/oidc"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+	"golang.org/x/xerrors"
+
+	"github.com/coder/coder/cli/cliui"
+	"github.com/coder/coder/cli/config"
+	"github.com/coder/coder/codersdk"
+)
+
+func newConfig() codersdk.DeploymentConfig {
+	return codersdk.DeploymentConfig{
+		AccessURL: codersdk.DeploymentConfigField[string]{
+			Key:   "access_url",
+			Usage: "External URL to access your deployment. This must be accessible by all provisioned workspaces.",
+			Flag:  "access-url",
+		},
+		WildcardAccessURL: codersdk.DeploymentConfigField[string]{
+			Key:   "wildcard_access_url",
+			Usage: "Specifies the wildcard hostname to use for workspace applications in the form \"*.example.com\".",
+			Flag:  "wildcard-access-url",
+		},
+		Address: codersdk.DeploymentConfigField[string]{
+			Key:       "address",
+			Usage:     "Bind address of the server.",
+			Flag:      "address",
+			Shorthand: "a",
+			Value:     "127.0.0.1:3000",
+		},
+		AutobuildPollInterval: codersdk.DeploymentConfigField[time.Duration]{
+			Key:    "autobuild_poll_interval",
+			Usage:  "Interval to poll for scheduled workspace builds.",
+			Flag:   "autobuild-poll-interval",
+			Hidden: true,
+			Value:  time.Minute,
+		},
+		DERPServerEnable: codersdk.DeploymentConfigField[bool]{
+			Key:   "derp.server.enable",
+			Usage: "Whether to enable or disable the embedded DERP relay server.",
+			Flag:  "derp-server-enable",
+			Value: true,
+		},
+		DERPServerRegionID: codersdk.DeploymentConfigField[int]{
+			Key:   "derp.server.region_id",
+			Usage: "Region ID to use for the embedded DERP server.",
+			Flag:  "derp-server-region-id",
+			Value: 999,
+		},
+		DERPServerRegionCode: codersdk.DeploymentConfigField[string]{
+			Key:   "derp.server.region_code",
+			Usage: "Region code to use for the embedded DERP server.",
+			Flag:  "derp-server-region-code",
+			Value: "coder",
+		},
+		DERPServerRegionName: codersdk.DeploymentConfigField[string]{
+			Key:   "derp.server.region_name",
+			Usage: "Region name that for the embedded DERP server.",
+			Flag:  "derp-server-region-name",
+			Value: "Coder Embedded Relay",
+		},
+		DERPServerSTUNAddresses: codersdk.DeploymentConfigField[[]string]{
+			Key:   "derp.server.stun_addresses",
+			Usage: "Addresses for STUN servers to establish P2P connections. Set empty to disable P2P connections.",
+			Flag:  "derp-server-stun-addresses",
+			Value: []string{"stun.l.google.com:19302"},
+		},
+		DERPServerRelayURL: codersdk.DeploymentConfigField[string]{
+			Key:        "derp.server.relay_url",
+			Usage:      "An HTTP URL that is accessible by other replicas to relay DERP traffic. Required for high availability.",
+			Flag:       "derp-server-relay-url",
```

Comment:
Are you ok with this breaking this flag?

Reply:
Yup, URL just makes more sense and this one is new